SparkConnector/sparkconnector/configuration.py

```diff
@@ -132,11 +132,27 @@ def configure(self, opts, ports):
 
         # Extend conf adding logging of log4j to java options
         base_extra_java_options = "-Dlog4j.configuration=file:%s" % self.connector.log4j_file
-        extra_java_options = conf.get("spark.driver.extraJavaOptions")
-        if extra_java_options:
-            extra_java_options = base_extra_java_options + " " + extra_java_options
+        extra_java_options = None
+
+        # For customenv sessions, do not allow extraJavaOptions to be overwritten by the extra_java_options, 
+        # since the NXCals instances, coming as a default value by the SparkContext when extraJavaOptions is
+        # not specified, need to be preserved in the driver configuration.
+        if os.environ.get("SOFTWARE_SOURCE") == "customenv":
+            SPARK_DEFAULTS = os.environ.get(
+                "SPARK_DEFAULTS", 
+                os.path.join(os.environ.get("SPARK_HOME"), "conf", "spark-defaults.conf")
+            )
+            if os.path.exists(SPARK_DEFAULTS):
+                for line in open(SPARK_DEFAULTS).readlines():
+                    if line.startswith("spark.driver.extraJavaOptions"):
+                        service_url = re.search(r'-Dservice\.url=[^\s]+', line).group(0)
```

Comment:
This only matches -Dservice.url= . But what if NXCALS change tomorrow the value of extraJavaOptions to include more options? You would miss them. You need to take everything that is defined as spark.driver.extraJavaOptions in that line.